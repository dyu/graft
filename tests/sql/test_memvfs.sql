-- Load the memvfs extension and open a new connection using it
-- The memvfs extension is located at crates/sqlite-plugin/examples/memvfs.rs
.load libmemvfs
.open main.db

.databases
.vfsinfo
.vfsname

CREATE TABLE t1(a, b);
INSERT INTO t1 VALUES(1, 2);
INSERT INTO t1 VALUES(3, 4);
SELECT * FROM t1;
pragma hello_vfs=1234;

select * from dbstat;

vacuum;
drop table t1;
vacuum;

select * from dbstat;
