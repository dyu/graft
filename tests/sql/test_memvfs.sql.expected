-- Load the memvfs extension and open a new connection using it
-- The memvfs extension is located at crates/sqlite-plugin/examples/memvfs.rs
.load libmemvfs
(27) open: path=Some("main.db"), opts=OpenOpts { flags: 262, kind: MainDb, mode: ReadWrite { create: Create }, delete_on_close: false }
(27) read: file=Some("main.db"), offset=0, len=100
.open main.db

(27) pragma: file=Some("main.db"), pragma=Pragma { name: "database_list", arg: None }
.databases
main: main.db r/w
.vfsinfo
vfs.zName      = "mem"
vfs.iVersion   = 3
vfs.szOsFile   = 56
vfs.mxPathname = 512
.vfsname

(27) lock: file=Some("main.db"), level=Shared
(27) access: path=main.db-journal, flags=Exists
(27) access: path=main.db-wal, flags=Exists
(27) file_size: file=Some("main.db")
(27) unlock: file=Some("main.db"), level=Unlocked
(27) lock: file=Some("main.db"), level=Shared
(27) access: path=main.db-journal, flags=Exists
(27) read: file=Some("main.db"), offset=24, len=16
(27) access: path=main.db-wal, flags=Exists
(27) file_size: file=Some("main.db")
(27) lock: file=Some("main.db"), level=Reserved
(27) pragma: file=Some("main.db"), pragma=Pragma { name: "integrity_check", arg: Some("t1") }
(27) lock: file=Some("main.db"), level=Exclusive
(27) write: file=Some("main.db"), offset=0, len=4096
(27) write: file=Some("main.db"), offset=4096, len=4096
(27) sync: file=Some("main.db")
(27) unlock: file=Some("main.db"), level=Shared
(27) unlock: file=Some("main.db"), level=Unlocked
CREATE TABLE t1(a, b);
(27) lock: file=Some("main.db"), level=Shared
(27) access: path=main.db-journal, flags=Exists
(27) read: file=Some("main.db"), offset=24, len=16
(27) access: path=main.db-wal, flags=Exists
(27) file_size: file=Some("main.db")
(27) lock: file=Some("main.db"), level=Reserved
(27) lock: file=Some("main.db"), level=Exclusive
(27) write: file=Some("main.db"), offset=0, len=4096
(27) write: file=Some("main.db"), offset=4096, len=4096
(27) sync: file=Some("main.db")
(27) unlock: file=Some("main.db"), level=Shared
(27) unlock: file=Some("main.db"), level=Unlocked
INSERT INTO t1 VALUES(1, 2);
(27) lock: file=Some("main.db"), level=Shared
(27) access: path=main.db-journal, flags=Exists
(27) read: file=Some("main.db"), offset=24, len=16
(27) access: path=main.db-wal, flags=Exists
(27) file_size: file=Some("main.db")
(27) lock: file=Some("main.db"), level=Reserved
(27) lock: file=Some("main.db"), level=Exclusive
(27) write: file=Some("main.db"), offset=0, len=4096
(27) write: file=Some("main.db"), offset=4096, len=4096
(27) sync: file=Some("main.db")
(27) unlock: file=Some("main.db"), level=Shared
(27) unlock: file=Some("main.db"), level=Unlocked
INSERT INTO t1 VALUES(3, 4);
(27) lock: file=Some("main.db"), level=Shared
(27) access: path=main.db-journal, flags=Exists
(27) read: file=Some("main.db"), offset=24, len=16
(27) access: path=main.db-wal, flags=Exists
(27) file_size: file=Some("main.db")
(27) unlock: file=Some("main.db"), level=Unlocked
SELECT * FROM t1;
+---+---+
| a | b |
+---+---+
| 1 | 2 |
| 3 | 4 |
+---+---+
(27) pragma: file=Some("main.db"), pragma=Pragma { name: "hello_vfs", arg: Some("1234") }
pragma hello_vfs=1234;

(27) lock: file=Some("main.db"), level=Shared
(27) access: path=main.db-journal, flags=Exists
(27) read: file=Some("main.db"), offset=24, len=16
(27) access: path=main.db-wal, flags=Exists
(27) file_size: file=Some("main.db")
(27) unlock: file=Some("main.db"), level=Unlocked
select * from dbstat;
+---------------+------+--------+----------+-------+---------+--------+------------+----------+--------+
|     name      | path | pageno | pagetype | ncell | payload | unused | mx_payload | pgoffset | pgsize |
+---------------+------+--------+----------+-------+---------+--------+------------+----------+--------+
| sqlite_schema | /    | 1      | leaf     | 1     | 37      | 3947   | 37         | 0        | 4096   |
| t1            | /    | 2      | leaf     | 2     | 9       | 4071   | 5          | 4096     | 4096   |
+---------------+------+--------+----------+-------+---------+--------+------------+----------+--------+

(27) lock: file=Some("main.db"), level=Shared
(27) access: path=main.db-journal, flags=Exists
(27) read: file=Some("main.db"), offset=24, len=16
(27) access: path=main.db-wal, flags=Exists
(27) file_size: file=Some("main.db")
(27) lock: file=Some("main.db"), level=Reserved
(27) lock: file=Some("main.db"), level=Exclusive
(27) write: file=Some("main.db"), offset=0, len=4096
(27) write: file=Some("main.db"), offset=4096, len=4096
(27) sync: file=Some("main.db")
(27) unlock: file=Some("main.db"), level=Shared
(27) unlock: file=Some("main.db"), level=Unlocked
vacuum;
(27) lock: file=Some("main.db"), level=Shared
(27) access: path=main.db-journal, flags=Exists
(27) read: file=Some("main.db"), offset=24, len=16
(27) access: path=main.db-wal, flags=Exists
(27) file_size: file=Some("main.db")
(27) unlock: file=Some("main.db"), level=Unlocked
(27) lock: file=Some("main.db"), level=Shared
(27) access: path=main.db-journal, flags=Exists
(27) read: file=Some("main.db"), offset=24, len=16
(27) access: path=main.db-wal, flags=Exists
(27) file_size: file=Some("main.db")
(27) lock: file=Some("main.db"), level=Reserved
(27) lock: file=Some("main.db"), level=Exclusive
(27) write: file=Some("main.db"), offset=0, len=4096
(27) write: file=Some("main.db"), offset=4096, len=4096
(27) sync: file=Some("main.db")
(27) unlock: file=Some("main.db"), level=Shared
(27) unlock: file=Some("main.db"), level=Unlocked
drop table t1;
(27) lock: file=Some("main.db"), level=Shared
(27) access: path=main.db-journal, flags=Exists
(27) read: file=Some("main.db"), offset=24, len=16
(27) access: path=main.db-wal, flags=Exists
(27) file_size: file=Some("main.db")
(27) lock: file=Some("main.db"), level=Reserved
(27) lock: file=Some("main.db"), level=Exclusive
(27) write: file=Some("main.db"), offset=0, len=4096
(27) sync: file=Some("main.db")
(27) file_size: file=Some("main.db")
(27) truncate: file=Some("main.db"), size=4096
(27) unlock: file=Some("main.db"), level=Shared
(27) unlock: file=Some("main.db"), level=Unlocked
vacuum;

(27) lock: file=Some("main.db"), level=Shared
(27) access: path=main.db-journal, flags=Exists
(27) read: file=Some("main.db"), offset=24, len=16
(27) access: path=main.db-wal, flags=Exists
(27) file_size: file=Some("main.db")
(27) unlock: file=Some("main.db"), level=Unlocked
(27) lock: file=Some("main.db"), level=Shared
(27) access: path=main.db-journal, flags=Exists
(27) read: file=Some("main.db"), offset=24, len=16
(27) access: path=main.db-wal, flags=Exists
(27) file_size: file=Some("main.db")
(27) unlock: file=Some("main.db"), level=Unlocked
select * from dbstat;
+---------------+------+--------+----------+-------+---------+--------+------------+----------+--------+
|     name      | path | pageno | pagetype | ncell | payload | unused | mx_payload | pgoffset | pgsize |
+---------------+------+--------+----------+-------+---------+--------+------------+----------+--------+
| sqlite_schema | /    | 1      | leaf     | 0     | 0       | 3988   | 0          | 0        | 4096   |
+---------------+------+--------+----------+-------+---------+--------+------------+----------+--------+
(27) unlock: file=Some("main.db"), level=Unlocked
(27) close: file=Some("main.db")

SQLite Exit Code = 0
